pyanaconda/ui/tui/spokes/software_selection.py

```diff
@@ -196,34 +196,38 @@ def refresh(self, args=None):
         threadMgr.wait(THREAD_CHECK_SOFTWARE)
         self._container = ListColumnContainer(2, columns_width=38, spacing=2)
 
-        # Display the environments
         if args is None:
-            environments = self.payload.environments
-            msg = _("Base environment")
+            msg = self._refresh_environments()
+        else:
+            msg = self._refresh_addons(args)
 
-            for env in environments:
-                name = self.payload.environmentDescription(env)[0]
-                selected = (env == self._selected_environment)
-                widget = CheckboxWidget(title="%s" % name, completed=selected)
-                self._container.add(widget, callback=self._set_environment_callback, data=env)
+        self.window.add_with_separator(TextWidget(msg))
+        self.window.add_with_separator(self._container)
 
-        # Display the add-ons
-        else:
-            length = len(args)
+    def _refresh_environments(self):
+        environments = self.payload.environments
 
-            if length > 0:
-                msg = _("Add-ons for selected environment")
-            else:
-                msg = _("No add-ons to select.")
+        for env in environments:
+            name = self.payload.environmentDescription(env)[0]
+            selected = (env == self._selected_environment)
+            widget = CheckboxWidget(title="%s" % name, completed=selected)
+            self._container.add(widget, callback=self._set_environment_callback, data=env)
 
-            for addon_id in args:
-                name = self.payload.groupDescription(addon_id)[0]
-                selected = addon_id in self._addons_selection
-                widget = CheckboxWidget(title="%s" % name, completed=selected)
-                self._container.add(widget, callback=self._set_addons_callback, data=addon_id)
+        return _("Base environment")
 
-        self.window.add_with_separator(TextWidget(msg))
-        self.window.add_with_separator(self._container)
+    def _refresh_addons(self, available_addons):
+        length = len(available_addons)
+
+        for addon_id in available_addons:
+            name = self.payload.groupDescription(addon_id)[0]
+            selected = addon_id in self._addons_selection
+            widget = CheckboxWidget(title="%s" % name, completed=selected)
+            self._container.add(widget, callback=self._set_addons_callback, data=addon_id)
+
+        if length > 0:
```

Comment:
This can be changed to `if available_addons` and `length` can be removed.

Reply:
And this is why it is important to polish code from time to time 👍 . Thanks!